src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -142,22 +142,30 @@ private void WriteDeltaResource(object graph, ODataWriter writer, ODataSerialize
             }
         }
 
-        private void WriteDeltaComplexProperties(IEnumerable<IEdmStructuralProperty> complexProperties,
+        private void WriteDeltaComplexProperties(SelectExpandNode selectExpandNode,
             ResourceContext resourceContext, ODataWriter writer)
         {
-            Contract.Assert(complexProperties != null);
             Contract.Assert(resourceContext != null);
             Contract.Assert(writer != null);
```

Comment:
Should we assert that selectExpandNode is not null?

Reply:
It's unnecessary, because it's a private method only called when the "SelectExpandNode != null".


---
In reply to: [352414171](https://github.com/OData/WebApi/pull/1941#discussion_r352414171) [](ancestors = 352414171)